Issues when deleting several realms at the same time

### Before reporting an issue

- [x] I have read and understood the above terms for submitting issues, and I understand that my issue may be closed without action if I do not follow them.

### Area

admin/api

### Describe the bug

Checking issue #46081, the root cause is the realms created for the test are deleted at once at the end. I see this exception in the server:

```
2026-02-10 12:26:11,530 ERROR [org.keycloak.services.error.KeycloakErrorHandler] (executor-thread-5) Uncaught server error: org.keycloak.models.ModelException: Database operation failed
	at org.keycloak.connections.jpa.support.EntityManagerProxy.convert(EntityManagerProxy.java:151)
	at org.keycloak.connections.jpa.support.EntityManagerProxy.invoke(EntityManagerProxy.java:113)
	at jdk.proxy2/jdk.proxy2.$Proxy55.find(Unknown Source)
	at org.keycloak.models.jpa.JpaRealmProvider.removeRealm(JpaRealmProvider.java:190)
	at org.keycloak.models.cache.infinispan.RealmCacheSession.removeRealm(RealmCacheSession.java:583)
	at org.keycloak.services.managers.RealmManager.removeRealm(RealmManager.java:298)
	at org.keycloak.services.resources.admin.RealmAdminResource.deleteRealm(RealmAdminResource.java:564)
	at org.keycloak.services.resources.admin.RealmAdminResource$quarkusrestinvoker$deleteRealm_b77a2ab5b922cfe955b15a076ab917e04c17b4f1.invoke(Unknown Source)
	at org.jboss.resteasy.reactive.server.handlers.InvocationHandler.handle(InvocationHandler.java:29)
	at io.quarkus.resteasy.reactive.server.runtime.QuarkusResteasyReactiveRequestContext.invokeHandler(QuarkusResteasyReactiveRequestContext.java:190)
	at org.jboss.resteasy.reactive.common.core.AbstractResteasyReactiveContext.run(AbstractResteasyReactiveContext.java:147)
	at io.quarkus.vertx.core.runtime.VertxCoreRecorder$15.runWith(VertxCoreRecorder.java:666)
	at org.jboss.threads.EnhancedQueueExecutor$Task.doRunWith(EnhancedQueueExecutor.java:2651)
	at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2630)
	at org.jboss.threads.EnhancedQueueExecutor.runThreadBody(EnhancedQueueExecutor.java:1622)
	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1589)
	at org.jboss.threads.DelegatingRunnable.run(DelegatingRunnable.java:11)
	at org.jboss.threads.ThreadLocalResettingRunnable.run(ThreadLocalResettingRunnable.java:11)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	at java.base/java.lang.Thread.run(Thread.java:1583)
Caused by: java.lang.NullPointerException: Cannot invoke "org.hibernate.engine.spi.EntityEntry.getLoadedState()" because "entry" is null
	at org.hibernate.sql.exec.internal.lock.TableLock.applyLoadedState(TableLock.java:336)
	at org.hibernate.sql.exec.internal.lock.TableLock$NonToOneResultHandler.applyResult(TableLock.java:295)
	at org.hibernate.sql.exec.internal.lock.TableLock.lambda$performActions$1(TableLock.java:268)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1596)
	at org.hibernate.sql.exec.internal.lock.TableLock.performActions(TableLock.java:263)
	at org.hibernate.sql.exec.internal.lock.FollowOnLockingAction.lambda$performPostAction$1(FollowOnLockingAction.java:141)
	at java.base/java.util.HashMap.forEach(HashMap.java:1429)
	at org.hibernate.sql.exec.internal.lock.FollowOnLockingAction.lambda$performPostAction$0(FollowOnLockingAction.java:140)
	at java.base/java.util.IdentityHashMap.forEach(IdentityHashMap.java:1375)
	at org.hibernate.sql.exec.internal.lock.FollowOnLockingAction.performPostAction(FollowOnLockingAction.java:121)
	at org.hibernate.sql.exec.internal.JdbcSelectWithActions.performPostAction(JdbcSelectWithActions.java:122)
	at org.hibernate.sql.exec.internal.JdbcSelectExecutorStandardImpl.doExecuteQuery(JdbcSelectExecutorStandardImpl.java:220)
	at org.hibernate.sql.exec.internal.JdbcSelectExecutorStandardImpl.executeQuery(JdbcSelectExecutorStandardImpl.java:100)
	at org.hibernate.sql.exec.spi.JdbcSelectExecutor.executeQuery(JdbcSelectExecutor.java:63)
	at org.hibernate.sql.exec.spi.JdbcSelectExecutor.list(JdbcSelectExecutor.java:137)
	at org.hibernate.loader.ast.internal.SingleIdLoadPlan.load(SingleIdLoadPlan.java:139)
	at org.hibernate.loader.ast.internal.SingleIdEntityLoaderStandardImpl.load(SingleIdEntityLoaderStandardImpl.java:74)
	at org.hibernate.loader.ast.internal.AbstractEntityBatchLoader.load(AbstractEntityBatchLoader.java:89)
	at org.hibernate.loader.ast.internal.AbstractEntityBatchLoader.load(AbstractEntityBatchLoader.java:52)
	at org.hibernate.loader.ast.internal.EntityBatchLoaderInPredicate.load(EntityBatchLoaderInPredicate.java:105)
	at org.hibernate.persister.entity.AbstractEntityPersister.doLoad(AbstractEntityPersister.java:3530)
	at org.hibernate.persister.entity.AbstractEntityPersister.load(AbstractEntityPersister.java:3519)
	at org.hibernate.event.internal.DefaultLoadEventListener.loadFromDatasource(DefaultLoadEventListener.java:576)
	at org.hibernate.event.internal.DefaultLoadEventListener.loadFromCacheOrDatasource(DefaultLoadEventListener.java:550)
	at org.hibernate.event.internal.DefaultLoadEventListener.load(DefaultLoadEventListener.java:534)
	at org.hibernate.event.internal.DefaultLoadEventListener.doLoad(DefaultLoadEventListener.java:520)
	at org.hibernate.event.internal.DefaultLoadEventListener.load(DefaultLoadEventListener.java:186)
	at org.hibernate.event.internal.DefaultLoadEventListener.lockAndLoad(DefaultLoadEventListener.java:470)
	at org.hibernate.event.internal.DefaultLoadEventListener.doOnLoad(DefaultLoadEventListener.java:97)
	at org.hibernate.event.internal.DefaultLoadEventListener.onLoad(DefaultLoadEventListener.java:55)
	at org.hibernate.event.service.internal.EventListenerGroupImpl.fireEventOnEachListener(EventListenerGroupImpl.java:149)
	at org.hibernate.internal.SessionImpl.fireLoadNoChecks(SessionImpl.java:1291)
	at org.hibernate.internal.SessionImpl.fireLoad(SessionImpl.java:1278)
	at org.hibernate.internal.SessionImpl.load(SessionImpl.java:1261)
	at org.hibernate.loader.internal.IdentifierLoadAccessImpl.doLoad(IdentifierLoadAccessImpl.java:178)
	at org.hibernate.loader.internal.IdentifierLoadAccessImpl.lambda$load$0(IdentifierLoadAccessImpl.java:164)
	at org.hibernate.loader.internal.IdentifierLoadAccessImpl.perform(IdentifierLoadAccessImpl.java:127)
	at org.hibernate.loader.internal.IdentifierLoadAccessImpl.load(IdentifierLoadAccessImpl.java:164)
	at org.hibernate.internal.SessionImpl.find(SessionImpl.java:2194)
	at org.hibernate.internal.SessionImpl.find(SessionImpl.java:2184)
	at org.hibernate.internal.SessionImpl.find(SessionImpl.java:2173)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
	at org.keycloak.connections.jpa.support.EntityManagerProxy.invoke(EntityManagerProxy.java:101)
	... 18 more
```

I don't know if something in hibernate or similar. But this was not seen before so I would bet to something in hibernate has changed and it's triggering this when deleting realms now.

### Version

999.0.0-SNAPSHOT

### Regression

- [ ] The issue is a regression

### Expected behavior

The realms were delete before without issues.

### Actual behavior

Now this exception is seen and it's making the Admin UI job for the CI flaky.

### How to Reproduce?

See previous issue #46081

### Anything else?

_No response_

--- a/pom.xml
+++ b/pom.xml
@@ -40,8 +40,8 @@
 
         <asciidoctor.plugin.version>1.5.8</asciidoctor.plugin.version>
 
-        <quarkus.version>3.33.1.1</quarkus.version>
-        <quarkus.build.version>3.33.1.1</quarkus.build.version>
+        <quarkus.version>3.33.2</quarkus.version>
+        <quarkus.build.version>3.33.2</quarkus.build.version>
         <jboss-logging-annotations.version>3.0.4.Final</jboss-logging-annotations.version> <!-- keep in sync with the version used by quarkus -->
 
         <project.build-time>${timestamp}</project.build-time>
@@ -100,7 +100,7 @@
         <jboss.spec.javax.xml.bind.jboss-jaxb-api_2.3_spec.version>2.0.1.Final</jboss.spec.javax.xml.bind.jboss-jaxb-api_2.3_spec.version>
         <jboss.spec.javax.servlet.jsp.jboss-jsp-api_2.3_spec.version>2.0.0.Final</jboss.spec.javax.servlet.jsp.jboss-jsp-api_2.3_spec.version>
         <log4j2-api.version>2.25.3</log4j2-api.version> <!-- Odd name needs to align with Quarkus -->
-        <resteasy.version>6.2.15.Final</resteasy.version>
+        <resteasy.version>6.2.16.Final</resteasy.version>
         <resteasy.undertow.version>${resteasy.version}</resteasy.undertow.version>
         <owasp.html.sanitizer.version>20260101.1</owasp.html.sanitizer.version>
         <slf4j.version>2.0.6</slf4j.version>
@@ -111,7 +111,7 @@
         <sun.activation.version>1.2.2</sun.activation.version>
         <org.glassfish.jaxb.xsom.version>2.3.3-b02</org.glassfish.jaxb.xsom.version>
         <undertow.version>2.3.24.Final</undertow.version>
-        <wildfly-elytron.version>2.8.3.Final</wildfly-elytron.version>
+        <wildfly-elytron.version>2.8.4.Final</wildfly-elytron.version>
         <elytron.undertow-server.version>1.9.0.Final</elytron.undertow-server.version>
         <woodstox.version>6.0.3</woodstox.version>
         <wildfly.common.quarkus.aligned.version>1.5.4.Final-format-001</wildfly.common.quarkus.aligned.version>

